library/core/src/main/java/com/google/android/exoplayer2/metadata/dvbsi/AitDecoder.java

```diff
@@ -0,0 +1,195 @@
+            int aitControlCode = -1;
+
+            long application_identifier = sectionData.readUnsignedInt24() << 24L;
+            application_identifier |= sectionData.readUnsignedInt24();
+            int controlCode = sectionData.readUnsignedByte();
+
+            aitControlCode = controlCode;
+
+            tmp = sectionData.readUnsignedShort();
+            int sectionLength = tmp & 4095;
+            int positionOfNextSection = sectionData.getPosition() + sectionLength;
+            while(sectionData.getPosition() < positionOfNextSection) {
+                int type = sectionData.readUnsignedByte();
+                int l = sectionData.readUnsignedByte();
+                int positionOfNextSection2 = sectionData.getPosition() + l;
+
+                if(type == DESCRIPTOR_TRANSPORT_PROTOCOL) {
+                    int protocolId = sectionData.readUnsignedShort();
+                    int label = sectionData.readUnsignedByte();
+
+                    if(protocolId == TRANSPORT_PROTOCOL_HTTP) {
+                        while (sectionData.getPosition() < positionOfNextSection2) {
+                            int urlBaseLength = sectionData.readUnsignedByte();
+                            String urlBase = sectionData.readString(urlBaseLength);
+                            int extensionCount = sectionData.readUnsignedByte();
+                            aitUrlBase = urlBase;
+                            for (int i = 0; i < extensionCount; i++) {
+                                int len = sectionData.readUnsignedByte();
+                                sectionData.skipBytes(len);
+                            }
+                        }
+                    }
+                } else if(type == DESCRIPTOR_SIMPLE_APPLICATION_LOCATION) {
+                    String url = sectionData.readString(l);
+                    aitUrlExtension = url;
+                }
+
+                sectionData.setPosition(positionOfNextSection2);
+            }
+
+            sectionData.setPosition(positionOfNextSection);
+
+            if(aitControlCode != -1 && aitUrlBase != null && aitUrlExtension != null) {
+                aits.add(new Ait(aitControlCode, aitUrlBase + aitUrlExtension));
+            }
+        }
+
+        return new Metadata(aits);
+    }
+
+    static final String dvbCharset[] = {
+        "ISO-8859-15",
+        "ISO-8859-5",
+        "ISO-8859-6",
+        "ISO-8859-7",
+        "ISO-8859-8",
+        "ISO-8859-9",
+        null,
+        "ISO-8859-11",
+        null,
+        "ISO-8859-13",
+        null,
+        "ISO-8859-15",
+        null,
+        null,
+        null,
+        null,
+        //0x10
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+    };
+
+    static String readDvbString(ParsableByteArray data, int length) {
```

Comment:
Is this method used? I couldn't see a reference to it

Reply:
Correct, it isn't used.
That was used by some dead code (there are many informations not parsed by this decoder) (namely to get the "Application name")

This will be removed, thanks